<launch>
<!-- This launch file is for policies trained with isaac lab and the model is exported to onnx -->
  <arg name="onnx_path" default="rl_games/kingfisher_direct.onnx"/>
  <arg name="simulation" default="true"/>
  <arg name="cmd_topic" default="/rl_agent/cmd_drive"/>
  <arg name="odom_topic" default="/local_odom"/>
  <arg name="goal_topic" default="/move_base_simple/goal"/>
  <arg name="status_topic" default="/agent_status"/>

  <node pkg="kingfisher_rl" type="capture_agent_onnx.py" name="rl_agent" output="screen">
    <remap from="~goal" to="$(arg goal_topic)"/>
    <remap from="~odom" to="$(arg odom_topic)"/> <!-- /imu/odometry | /pose_gt -->
    <remap from="~rl_status" to="$(arg status_topic)"/>
    <remap from="~cmd_drive" to="$(arg cmd_topic)"/>

    <param name="onnx_path" value="$(arg onnx_path)"/>
    <param name="dist_threshold" value="0.3"/>
    <param name="control_freq" value="20"/>
    <param name="obs_buffer_len" value="1"/>
  </node>

</launch>
